lib/src/components/NxRadio/__tests__/NxRadio.test.tsx

```diff
@@ -21,146 +22,131 @@ describe('NxRadio', function() {
+
+    await user.click(screen.getByRole('radio'));
 
-    component.find('input').simulate('change');
     expect(onChange).toHaveBeenCalledWith(null);
   });
 
   it('passes unknown props to the label element', function() {
-    const component = getShallow({ id: 'foo', htmlFor: 'baz' });
+    const el = renderEl({ id: 'foo', htmlFor: 'baz' });
 
-    expect(component).toHaveProp('id', 'foo');
-    expect(component).toHaveProp('htmlFor', 'baz');
+    expect(el).toHaveAttribute('id', 'foo');
+    expect(el).toHaveAttribute('for', 'baz');
   });
 
   describe('when disabled prop is true', function () {
     it('adds the nx-radio-checkbox--disabled class', function() {
-      expect(getShallow()).not.toHaveClassName('nx-radio-checkbox--disabled');
-      expect(getShallow({ disabled: true })).toHaveClassName('nx-radio-checkbox--disabled');
+      expect(renderEl()).not.toHaveClass('nx-radio-checkbox--disabled');
+      expect(renderEl({ disabled: true })).toHaveClass('nx-radio-checkbox--disabled');
     });
 
     it('disables the input', function() {
-      expect(getShallow().find('input')).toHaveProp('disabled', false);
-      expect(getShallow({ disabled: false }).find('input')).toHaveProp('disabled', false);
-      expect(getShallow({ disabled: true }).find('input')).toHaveProp('disabled', true);
+      expect(renderEl()?.querySelector('input')).not.toHaveAttribute('disabled');
+      expect(renderEl({ disabled: false })?.querySelector('input')).not.toHaveAttribute('disabled');
+      expect(renderEl({ disabled: true })?.querySelector('input')).toHaveAttribute('disabled');
     });
   });
 
   describe('isChecked prop', function () {
     it('sets the input to checked per the value of isChecked', function() {
-      expect(getShallow({ isChecked: false }).find('input')).toHaveProp('checked', false);
-      expect(getShallow({ isChecked: true }).find('input')).toHaveProp('checked', true);
+      expect(renderEl({ isChecked: false })?.querySelector('input')).not.toHaveAttribute('checked');
+      expect(renderEl({ isChecked: true })?.querySelector('input')).toHaveAttribute('checked');
     });
 
     it('adds the tm-checked class if isChecked is true, and the tm-unchecked class if it is false', function() {
-      expect(getShallow()).toHaveClassName('tm-unchecked');
-      expect(getShallow()).not.toHaveClassName('tm-checked');
+      expect(renderEl()).toHaveClass('tm-unchecked');
+      expect(renderEl()).not.toHaveClass('tm-checked');
 
-      expect(getShallow({ isChecked: true })).not.toHaveClassName('tm-unchecked');
-      expect(getShallow({ isChecked: true })).toHaveClassName('tm-checked');
+      expect(renderEl({ isChecked: true })).not.toHaveClass('tm-unchecked');
+      expect(renderEl({ isChecked: true })).toHaveClass('tm-checked');
     });
   });
 
   describe('children prop', function () {
     it('renders children nodes within an nx-radio__content <span>', function() {
```

Comment:
The `span.nx-radio__content` itself is an implementation detail, this test should be simplified to just check that `.bar` is among the descendants.

Reply:
> The `span.nx-radio__content` itself is an implementation detail, this test should be simplified to just check that `.bar` is among the descendants.

Updated here: 2c79beb07760dd08ceb80320a591cf9689044f2d